src/main/ml-modules/root/lib/redaction/redactionUtils.sjs

```diff
@@ -0,0 +1,487 @@
+const LOCK = Symbol('lock');
```

Comment:
Looks like solid code, but the reader (or me at least) has a hard time figuring out how the ciphers work. At the end of the day a couple key things should be more clear

* Every type of cypher has a set of chars that it applies to. Other chars are left as-is to preserve formatting.
* The basic cipher algorithm is to choose an array of offsets and use the same offset for the Nth character in the input string. Where an "offset" is the number of characters or digits to advance the input char (not sure if this is numeric, or offset into the allowed chars array)

Comments may clear a lot up, but fewer levels of indirection may also make it easier to use and extend the library.

Some distinction between deterministic and non-colliding should be made. The ciphers are all non-colliding, I think. If collisions are ok, a more basic random value will be more secure, at the cost of collisions.